app/models/tag.rb

```diff
@@ -130,6 +130,85 @@ def errors_as_sentence
     errors.full_messages.to_sentence
   end
 
+  # @param follower [#id, #class_name] An object who's class "acts_as_follower" (e.g. a User).
+  # @return [ActiveRecord::Relation<Tag>] with the "points" method added.
+  #
+  # @note This method will also add the follower's "points" for the given tag.  In the
+  #       ActiveRecord::Base implementation, we can add "virtual" attributes by including them in
+  #       the select statement (as shown in the method implementation).  Doing this can sometimes
+  #       result in a surprise, so you may want to consider casting the results into a well-defined
+  #       data structure.  But then you might be looking at implementing the DataMapper pattern.
+  #
+  #
+  # @example
+  #   Below is the SQL generated:
+  #
+  #   ```sql
+  #     SELECT tags.*, "followings"."points"
+  #       FROM "tags"
+  #         INNER JOIN "follows" "followings"
+  #           ON "followings"."followable_type" = 'ActsAsTaggableOn::Tag'
+  #           AND "followings"."followable_id" = "tags"."id"
+  #       WHERE "followings"."follower_id" = 1
+  #          AND "followings"."follower_type" = 'User'
+  #     ORDER BY "tags"."hotness_score" DESC
+  #   ```
+  #
+  # @see Tag#points
+  #
+  # @see UserDecorator::CACHED_TAGGED_BY_USER_ATTRIBUTES for discussion on why we're selecting this.
+  #
+  # @todo should we sort by hotness score?  Wouldn't the user's points make more sense?
+  def self.cached_followed_tags_for(follower:)
+    Tag
+      .select(
+        "tags.bg_color_hex",
+        "tags.hotness_score",
+        "tags.id",
+        "tags.name",
+        "tags.text_color_hex",
+        "followings.points",
+      )
+      .joins(:followings)
+      .where("followings.follower_id" => follower.id, "followings.follower_type" => follower.class_name)
+      .order(hotness_score: :desc)
+  end
```

Comment:
I'm a bit confused by the name of this method. Are we using "cached" to mean something other than what I thought it meant?

Reply:
Oh, I see, we're calling it inside of a cache block. But there's nothing about this method itself that does caching. Is there a way to make this clearer?

Reply:
I removed the `cached_` prefix and added documentation.  The other option is to move this out of the Tag class; I'm about 70% against moving it, but open for suggestions.

Reply:
I am aiming for a modicum of performance increase.  It's also possible we would no longer need to cache this on the decorator (instead relying on the JSON document caching for 15 minutes).  But that's an exercise for a future reader.

Reply:
Okay, is this in the `/dashboard/following_tags` route? Just want to be able to track the performance difference.

Reply:
It's in the `/async_info/base_data` and a callback for calculating an article's score.